src/js/requestHandlers.js

```diff
@@ -230,6 +235,85 @@ sjrk.storyTelling.server.deleteStoryFromCouch = function (id, deleteStoryDataSou
     return promise;
 };
 
+sjrk.storyTelling.server.deleteStoryFiles = function (storyContent, serverConfig) {
+    var filesToDelete = [];
+
+    fluid.each(storyContent, function (block) {
+        var blockFileName = "";
+
+        if (block.blockType === "image") {
+            blockFileName = block.imageUrl;
+        } else if (block.blockType === "audio" || block.blockType === "video") {
+            blockFileName = block.mediaUrl;
+        }
+
+        if (sjrk.storyTelling.server.isValidMediaFilename(blockFileName)) {
+            filesToDelete.push(blockFileName);
+        }
+    });
+
+    // remove duplicate entries so we don't try to delete already-deleted files
+    filesToDelete = filesToDelete.filter(function (fileName, index, self) {
+        return self.indexOf(fileName) === index;
+    });
+
+    fluid.each(filesToDelete, function (fileToDelete) {
+        sjrk.storyTelling.server.deleteSingleFileRecoverable(fileToDelete, serverConfig);
+    });
+};
+
+sjrk.storyTelling.server.isValidMediaFilename = function (fileName, serverConfig) {
+    try {
+        if (!fileName || fileName === "null" || fileName === "undefined")
+        {
+            throw "The filename is not valid: not truthy.";
```

Comment:
This structure is overly complex: to throw an exception only to catch it within the same function and return `false` -- let's instead just return `false` directly from these tests.

Reply:
Are these tests against "null" and "undefined" necessary? Won't they fail the regular expression test anyway?